TwilioVerifyDemo/NotificationExtension/NotificationService.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  NotificationService.swift
+//  TwilioVerifyDemo
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import UserNotifications
+import TwilioVerifySDK
+import TwilioVerifyDemoCache
+
+class NotificationService: UNNotificationServiceExtension {
+  
+  var contentHandler: ((UNNotificationContent) -> Void)?
+  var bestAttemptContent: UNMutableNotificationContent?
+  
+  override func didReceive(
+    _ request: UNNotificationRequest,
+    withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void
+  ) {
+    self.contentHandler = contentHandler
+    bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent)
+    
+    if let bestAttemptContent = bestAttemptContent {
+      // Modify the notification content here...
+      bestAttemptContent.title = "\(bestAttemptContent.title) [modified]"
+      
+      // TODO: Retrieve challenge from TwilioSDK and save it on TwilioVerifyDemoCache
```

Comment:
<table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/64d430897a2df4dbbabee1ff15ea40919450eaae/TwilioVerifyDemo/NotificationExtension/NotificationService.swift#L38"></span>TODOs should be resolved (Retrieve challenge from Twilio…).<br />
<code>todo</code> <code>NotificationService.swift:38</code></td>
    </tr>
  </tbody>
</table>